sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,544 @@
+    print(document.errors)
+
+print("Form type ID: {}".format(train_result.form_type_id))
+print("Extracted fields:")
+print(train_result.extracted_fields)
+
+# Analyze
+blob_sas_url = "xxxxx"  # form to analyze uploaded to blob storage
+poller = client.begin_extract_forms(blob_sas_url, model_id=model.model_id)
+result = poller.result()
+
+for page in result:
+    print("Page: {}".format(page.page_number))
+    print("Form type ID: {}".format(page.form_type_id))
+    for field in page.fields:
+        print("{}: {}".format(field.name.text, field.value.text))
+        print("Confidence: {}".format(field.confidence))
+```
+
+#### Train and Analyze with labels
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint=endpoint, credential=credential)
+
+# Train
+blob_sas_url = "xxxxx"  # training documents uploaded to blob storage
+poller = client.begin_labeled_training(blob_sas_url)
+model = poller.result()
+
+# Custom model information
+print("Model ID: {}".format(model.model_id))
+print("Status: {}".format(model.status))
+print("Created on: {}".format(model.created_on))
+print("Last updated on: {}".format(model.last_updated_on))
+
+# Training result information
+train_result = model.train_result
+for document in train_result.documents:
+    print(document.name)
+    print(document.status)
+    print(document.page_count)
+    print(document.errors)
+
+print("Average model accuracy: {}".format(train_result.average_model_accuracy))
+print("Fields extracted/accuracy")
+for field in train_result.fields:
+    print(field.field_name, field.accuracy)
+
+blob_sas_url = "xxxxx"  # form to analyze uploaded to blob storage
+poller = client.begin_extract_labeled_forms(blob_sas_url, model_id=model.model_id)
+result = poller.result()
+
+print("Page range: {}".format(result.page_range))
+for field, value in result.labels.items():
+    print("{}: {}".format(field, value.text))
+    print(value.bounding_box, value.confidence)
+
+```
+
+#### List custom models
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint=endpoint, credential=credential)
+custom_models = list(client.list_custom_models())
+for model in custom_models:
+    print(model.model_id, model.status)
+```
+
+#### Get models summary
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint=endpoint, credential=credential)
+summary = client.get_models_summary()
+
+print("Number of models: {}".format(summary.count))
+print("Max number of models that can be trained with this subscription: {}".format(summary.limit))
+print("Datetime when summary was updated: {}".format(summary.last_updated_on))
+```
+
+
+#### Get a custom model with a model ID (e.g. unsupervised)
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint=endpoint, credential=credential)
+model = client.get_custom_model(model_id="xxxxx")
```

Comment:
It looks like this returns a `CustomModel` (no labels). Would there be another method to retrieve a `CustomLabeledModel`?

Reply:
Also since we are in a "Custom" Form Client, do we need the qualifier "custom" in `get_custom_model`? Same question applies to delete below.

Reply:
Yes, there is a `get_custom_labeled_model() -> CustomLabeledModel` method, I added a sample for clarity. I'll reconsider naming with Anne. My only consideration here is making sure the method name is descriptive enough that it can be searched for effectively.